torch/utils/data/_utils/collate.py

```diff
@@ -201,6 +202,39 @@ def collate_str_fn(batch, *, collate_fn_map: Optional[Dict[Union[Type, Tuple[Typ
 default_collate_fn_map[bytes] = collate_str_fn
 
 
+def register_default_collate_for(name: Type, collate_fn: Optional[Callable] = None) -> Callable:
```

Comment:
Why does `collate_fn` defaults to `None`? What is the expected behavior if it were `None`?

Reply:
This is because when using the decorator context, the `collate_fn` is not passed in as an argument, but will be included in the lambda expression. 
Currently if you does not pass a collate_fn when calling it, nothing will happen. I'm gonna make it complain a ValueError if no collate_fn is passed. 
Thanks for pointing out!